src/Features/Core/Portable/Completion/CompletionItemRules.cs

```diff
@@ -47,18 +70,48 @@ public sealed class CompletionItemRules
         /// </summary>
         public int MatchPriority { get; }
 
+        /// <summary>
+        /// How this item should be selected when the completion list first appears and
+        /// before the user has typed any characters.
+        /// </summary>
+        public CompletionItemSelectionBehavior SelectionBehavior { get; }
+
         private CompletionItemRules(
             ImmutableArray<CharacterSetModificationRule> filterCharacterRules,
             ImmutableArray<CharacterSetModificationRule> commitCharacterRules,
             EnterKeyRule enterKeyRule,
             bool formatOnCommit,
-            int matchPriority)
+            int matchPriority,
+            CompletionItemSelectionBehavior selectionBehavior)
+        {
+            FilterCharacterRules = filterCharacterRules.IsDefault ? ImmutableArray<CharacterSetModificationRule>.Empty : filterCharacterRules;
+            CommitCharacterRules = commitCharacterRules.IsDefault ? ImmutableArray<CharacterSetModificationRule>.Empty : commitCharacterRules;
+            EnterKeyRule = enterKeyRule;
+            FormatOnCommit = formatOnCommit;
+            MatchPriority = matchPriority;
+            SelectionBehavior = selectionBehavior;
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="CompletionItemRules"/> instance.
+        /// </summary>
+        /// <param name="filterCharacterRules">Rules about which keys typed are used to filter the list of completion items.</param>
+        /// <param name="commitCharacterRules">Rules about which keys typed caused the completion item to be committed.</param>
+        /// <param name="enterKeyRule">Rule about whether the enter key is passed through to the editor after the selected item has been committed.</param>
+        /// <param name="formatOnCommit">True if the modified text should be formatted automatically.</param>
+        /// <param name="matchPriority">True if the related completion item should be initially selected.</param>
+        /// <returns></returns>
+        public static CompletionItemRules Create(
+            ImmutableArray<CharacterSetModificationRule> filterCharacterRules,
+            ImmutableArray<CharacterSetModificationRule> commitCharacterRules,
+            EnterKeyRule enterKeyRule,
+            bool formatOnCommit,
+            int? matchPriority)
```

Comment:
This is the existing factory constructor, now without optional parameters, which now forwards to the new factory constructor.